Microsoft.Research/Contracts/MsCorlib/System.Math.cs

```diff
@@ -93,29 +93,40 @@ public static Decimal Abs(Decimal value)
     [Pure]
     public static float Abs(float value)
     {
+      Contract.Ensures((Contract.Result<float>() >= 0.0) || float.IsNaN(Contract.Result<float>()) || float.IsPositiveInfinity(Contract.Result<float>()));
```

Comment:
I think the checker had a problem to infer IsPositiveInfinity from x >= 0; if s.o. requires ```x >= 0 && !IsInfinity(x)```, this could pass if we not explicitly include IsPositiveInfinity!

I will add some test cases once I fully understood the CC regressing testing framework; once we have test cases, we can optimize this and maybe fix the checker to handle double/float more consistently.